ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -91,10 +92,11 @@ data FetchMode =
   deriving (Eq, Show)
 
 
-type PeerInfo header extra =
+type PeerInfo header peer extra =
```

Comment:
What about changing this into a record with strict fields.

Reply:
I'm not a fan of the "swizzel" functions used to access the different fields of the PeerInfo type, but I think that such a change is outside the scope of this PR.